### PR TITLE
libobs,UI: Replace and deprecate obs_scene_sceneitem_from_source

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

`obs_scene_sceneitem_from_source` is problematic because there can be multiple items of the same source in a scene, which the function doesn't account for. In such a case, it would return the first item it finds, which often might not be what a developer expects. 
It was originally added for the undo/redo-operation of "Add New Source" where the UI guarantees that the item is unique, but for a general case it's not suitable.

In the one UI use, instead of first creating the scene item and then finding it again , we can just remember the scene item in AddNew after we create it, similar to the source itself.

### Motivation and Context

The existence of the method was discussed on Discord, and I think it shouldn't exist.

### How Has This Been Tested?

macOS 15
Added and undo/redo'd (redid?) new sources without problems.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/window-basic-source-select.cpp [hunk 1]
```diff
@@ -340,12 +346,10 @@ void OBSBasicSourceSelect::on_buttonBox_accepted()
...
 				obs_data_create_from_json(data.c_str());
 			OBSSource source;
+			OBSSceneItem item;
 			AddNew(NULL, obs_data_get_string(dat, "id"),
 			       obs_data_get_string(dat, "name"),
-			       obs_data_get_bool(dat, "visible"), source);
-			OBSSceneItemAutoRelease item =
-				obs_scene_sceneitem_from_source(
-					main->GetCurrentScene(), source);
+			       obs_data_get_bool(dat, "visible"), source, item);
```
Comment (Lain-B) [inline UI/window-basic-source-select.cpp:352]: I'd prefer using `!!` to be more explicit that this is a conversion to boolean.

**Review by Lain-B — CHANGES_REQUESTED** [review]

File: UI/window-basic-source-select.cpp (hunk 1 above)
Comment (gxalpha) [reply UI/window-basic-source-select.cpp]: Changed.

Comment (Lain-B) [reply UI/window-basic-source-select.cpp]: Or not item, but `source`. `source` is being used as a boolean for the `visible` parameter isn't it?

Comment (Lain-B) [reply UI/window-basic-source-select.cpp:352]: Ah I misread due to the way Github shows the diff. My apologies.

**Review by Lain-B — APPROVED** [review]

**Pull request merged**